api/types/container/host_config.go

```diff
@@ -284,6 +284,17 @@ type DeviceMapping struct {
 	CgroupPermissions string
 }
 
+// HugepageLimit corresponds to the file`hugetlb.<hugepagesize>.limit_in_byte` in container level cgroup.
+// For example, `PageSize=1GB`, `Limit=1073741824` means setting `1073741824` bytes to hugetlb.1GB.limit_in_bytes.
+type HugepageLimit struct {
+	// The value of PageSize has the format <size><unit-prefix>B (2MB, 1GB),
+	// and must match the <hugepagesize> of the corresponding control file found in `hugetlb.<hugepagesize>.limit_in_bytes`.
+	// The values of <unit-prefix> are intended to be parsed using base 1024("1KB" = 1024, "1MB" = 1048576, etc).
+	PageSize string
```

Comment:
We should have some validation on this string.

Reply:
@cpuguy83 
@admanV have added a validation logic on here:
https://github.com/moby/moby/pull/40160/files#diff-b56258ecab11f7b797306907327cc904R574
Can you take a look changes?